Firebase.cpp

```diff
@@ -15,100 +15,158 @@
 //
 #include "Firebase.h"
 
-const char* firebaseFingerprint = "7A 54 06 9B DC 7A 25 B3 86 8D 66 53 48 2C 0B 96 42 C7 B3 0A";
-const uint16_t firebasePort = 443;
+namespace {
+const char* kFirebaseFingerprint = "7A 54 06 9B DC 7A 25 B3 86 8D 66 53 48 2C 0B 96 42 C7 B3 0A";
+const uint16_t kFirebasePort = 443;
+}  // namespace
 
-Firebase::Firebase(const String& host) : _host(host) {
-  _http.setReuse(true);
+Firebase::Firebase(const String& host) : connection_(host) {
 }
 
 Firebase& Firebase::auth(const String& auth) {
-  _auth = auth;
+  connection_.auth(auth);
   return *this;
 }
 
-String Firebase::get(const String& path) {
-  return sendRequest("GET", path);
+FirebaseResult Firebase::get(const String& path) {
```

Comment:
what do you think of storing the last result as a member of `Firebase`?
That way we could make reading/decoding optional.

```
fbase.post("/foo", "bar"); // just post the body
auto& result = fbase.post("/foo", "bar").json(); // get the result and decode it.
```